src/test/java/org/opensearch/security/test/AbstractSecurityUnitTest.java

```diff
@@ -278,6 +278,18 @@ public Settings get(int i) {
         };
     }
 
+    protected NodeSettingsSupplier genericMinimumSecuritySettings(List<Settings> others, List<Boolean> sslOnly) {
+
+        return i -> {
```

Comment:
this code is hard to understand without comments. Should it check size of `sslOnly` and `others` instead?

Reply:
Can check for overflow and underflow explicitly and then set to default value. Thoughts?